### PR TITLE
Support Universal Dependencies and bug fixes

### LINKED ISSUE #36: support universal dependencies
in Mr. Hug story:
     34 case
     30 nmod
     21 compound
      5 nummod
      4 nmod:poss
      4 acl:relcl
      1 nmod:tmod
      1 nmod:npmod
      1 compound:prt
      1 acl

@gaebor fyi

### LINKED ISSUE #104: if headword is present in definition, return definition's graph as is
e.g. intelligent
this might be enough:

if word in word2machine:                                                
    return word2machine[word]

### LINKED ISSUE #105: flag in lemmatze fnc to convert function words to uppercase
also move "have to HAS" hack to lemmatize

### PR DESCRIPTION
**Pull request by Eszti**
fixes #105 
fixes #104 
fixes #103 
fixes #36

### CODE REVIEW

File: src/fourlang/dep_to_4lang.py [hunk 1]
```diff
@@ -244,6 +247,13 @@ def create_from_line(line):
             # logging.info('adding new rel from: {0}'.format(dep))
             rel = dep.split('_', 1)[1].upper()
 
+        # Universal Dependencies
+        if ((dep.startswith('acl:') and not dep.startswith('acl:relcl')) or
+                dep.startswith('advcl:') or
+                dep.startswith('nmod:')) and rel is None:
+            logging.info('adding new rel from: {0}'.format(dep))
+            rel = dep.split(':', 1)[1].upper()
```
Comment (recski) [inline src/fourlang/dep_to_4lang.py]: @Eszti can you maybe show 1-2 examples for this case in the comments? Just so we'll know in the future what this is about

File: src/fourlang/lexicon.py [hunk 2]
```diff
@@ -179,23 +172,23 @@ def get_expanded_definition(self, printname):
...
         if not printname:
             return self.get_machine("_empty_")
 
         if printname.isupper():
-            return self.get_new_machine(printname)
-
-        # TODO: hack
-        if printname == 'have':
-            return self.get_machine('HAS')
+            return self.get_machine(printname, new_machine=True)
```
Comment (recski) [inline src/fourlang/lexicon.py]: should this ever happen now? how?

File: src/fourlang/similarity.py [hunk 3]
```diff
@@ -98,6 +98,7 @@ def lemma_similarities(self, lemma1, lemma2):
     def word_similarities(self, word1, word2):
         if (word1, word2) in self.word_sim_cache:
             return self.word_sim_cache[(word1, word2)]
+        # TODO: uppercase flag = ?
```
Comment (recski) [inline src/fourlang/similarity.py]: let's discuss these tomorrow

Comment (recski) [inline src/fourlang/similarity.py]: let's add the upper flag.

File: src/fourlang/dep_to_4lang.py (hunk 1 above)
Comment (Eszti) [inline src/fourlang/dep_to_4lang.py]: E.g.:
word: absenteeism
deps: 
"nmod:without(absence-2, reason-10)" => absenteeism <1- WITHOUT -2> reason
"nmod:from(absence-2, school-6)" => absenteeism <1- FROM -2> school

is it ok this way, or should I add comments into the code?

Comment (recski) [inline src/fourlang/dep_to_4lang.py]: this is fine, thanks

**Comment by recski** [comment]
Great!

**Pull request merged**